cmd/create.go

```diff
@@ -15,26 +15,34 @@ import (
 )
 
 func init() {
-	root.AddCommand(createCmd)
-	createCmd.Flags().BoolP("confirm", "c", false, "Prompt to confirm all configuration options (Env: $FUNC_CONFIRM)")
-	createCmd.Flags().StringP("runtime", "l", fn.DefaultRuntime, "Function runtime language/framework. Available runtimes: "+buildpacks.Runtimes()+" (Env: $FUNC_RUNTIME)")
-	createCmd.Flags().StringP("repositories", "r", filepath.Join(configPath(), "repositories"), "Path to extended template repositories (Env: $FUNC_REPOSITORIES)")
-	createCmd.Flags().StringP("template", "t", fn.DefaultTemplate, "Function template. Available templates: 'http' and 'events' (Env: $FUNC_TEMPLATE)")
+	// Add to the root a new "Create" command which obtains an appropriate
+	// instance of fn.Client from the given client creator function.
+	root.AddCommand(NewCreateCmd(newCreateClient))
+}
 
-	if err := createCmd.RegisterFlagCompletionFunc("runtime", CompleteRuntimeList); err != nil {
-		fmt.Println("internal: error while calling RegisterFlagCompletionFunc: ", err)
-	}
+// newCreateClient returns an instance of fn.Client for the "Create" command.
+// The createClientFn is a client factory which creates a new Client for use by
+// the create command during normal execution (see tests for alternative client
+// factories which return clients with various mocks).
+func newCreateClient(repositories string, verbose bool) *fn.Client {
+	return fn.New(fn.WithRepositories(repositories), fn.WithVerbose(verbose))
 }
 
-var createCmd = &cobra.Command{
-	Use:   "create [PATH]",
-	Short: "Create a function project",
-	Long: `Create a function project
+// createClientFn is a factory function which returns a Client suitable for
+// use with the Create command.
+type createClientFn func(repositories string, verbose bool) *fn.Client
+
+// NewCreateCmd creates a create command using the given client creator.
+func NewCreateCmd(clientFn createClientFn) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [PATH]",
+		Short: "Create a function project",
+		Long: `Create a function project
 
 Creates a new function project in PATH, or in the current directory if no PATH is given. 
 The name of the project is determined by the directory name the project is created in.
 `,
-	Example: `
+		Example: `
```

Comment:
I wonder if your editor is auto formatting this perhaps?

Reply:
Yes, this is `gofmt` fixing indentation to reflect that the struct is now embedded in a function.  Looks strange because of the string literals above, but it should be correct.